app/src/main/java/org/oppia/android/app/profileprogress/ProfileProgressActivity.kt

```diff
@@ -23,19 +28,40 @@ class ProfileProgressActivity :
   lateinit var profileProgressActivityPresenter: ProfileProgressActivityPresenter
   private var internalProfileId = -1
 
+  @Inject
+  lateinit var activityRouter: ActivityRouter
+
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     (activityComponent as ActivityComponentImpl).inject(this)
     internalProfileId = intent.getIntExtra(PROFILE_ID_EXTRA_KEY, -1)
     profileProgressActivityPresenter.handleOnCreate(internalProfileId)
   }
 
-  override fun routeToRecentlyPlayed() {
-    startActivity(
-      RecentlyPlayedActivity.createRecentlyPlayedActivityIntent(
-        this,
-        internalProfileId
-      )
+  override fun routeToRecentlyPlayed(title: String) {
+    val recentlyPlayedActivityParams =
+      RecentlyPlayedActivityParams
+        .newBuilder()
+        .setProfileId(ProfileId.newBuilder().setInternalId(internalProfileId).build())
+        .setActivityTitle(
+          when (title) {
+            getString(R.string.stories_for_you) -> {
```

Comment:
Similar to `getHeader()` function from `PromotedStoryListViewModel.kt` you should use `resourceHandler` to fetch strings here and everywhere else so that you can get translated strings.